### ISSUE TITLE
Drawer inside (native) Stack hangs when rapidly partially closed and opened

### ISSUE THREAD

**Issue by mhoran** [react-navigation/react-navigation#12347]
### Current behavior

When nesting a drawer (not a drawer navigator, but the extracted drawer layout) inside a native stack on iOS, rapidly closing the drawer partly, then reopening many times will eventually hang when letting go of the drawer. This only seems to happen when a drawer is nested in a stack navigator; when removing the stack I cannot reproduce.

### Expected behavior

The drawer should not hang. This is a bit of an edge case, but it could happen if the drawer is accidentally closed and rapidly opened again.

### Reproduction

https://github.com/mhoran/react-native-scratch/tree/drawer-layout-hang

### Platform

- [ ] Android
- [x] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [x] @react-navigation/native-stack
- [x] react-native-drawer-layout
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |^7.0.13
| @react-navigation/native-stack         |^7.1.14
| react-native-drawer-layout             |4.1.0
| react-native-screens                   |~4.1.0
| react-native-safe-area-context         |5.0.0
| react-native-gesture-handler           |^2.21.2
| react-native-reanimated                |~3.16.3
| react-native                           |0.76.3
| node                                   |v22
| npm or yarn                            |10.9.0

**Comment by mhoran** [comment]
The repro works fine. The issue still exists with latest dependencies.

**Comment by satya164** [comment]
> The repro works fine. The issue still exists with latest dependencies.

> **Don't link to a branch or specific file etc.** as it won't be detected.

**Comment by mhoran** [comment]
FWIW I determined that this issue is only reproducible when `react-native-screens` is loaded. It ultimately looks to be some interaction between `react-native-screens` and `react-native-reanimated`. I initially resolved the issue in my app by removing `react-native-screens` and using `react-navigation/stack` instead of `react-navigation/native-stack`.

There are some similar looking "hang" issues open against `react-native-screens`. Since I have found a workaround, I am not looking into this any further at this time.
